lib/routemaster/resources/rest_resource.rb

```diff
@@ -37,8 +37,9 @@ def update(id=nil, params)
       end
 
       def destroy(id=nil)
-        # no response wrapping as DELETE is supposed to 204.
-        @client.delete(@url.gsub('{id}', id.to_s))
+        @client.with_response(Responses::HateoasResponse) do
```

Comment:
I see, for me it was mostly about consistently getting the same response, but as we delegate most of the methods to `Faraday::Response` (including the response body) I really don't mind going in either way (it was just a proposal anyway).

I'll change it back to be a pure faraday response 👍